scripts/argocd-store-token.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+#
+# Uses the ArgoCD credentials to create a secret in the cluster. The credentials
+# are stored in a "env file" format sourced by this script, and later on using
+# "kubectl" to create the secret.
+#
+
+shopt -s inherit_errexit
+set -Eeu -o pipefail
+
+# Target secret name, to be created with ArgoCD credentials.
+declare -r SECRET_NAME="${SECRET_NAME:-rhtap-argocd-integration}"
+# Secret's namespace.
+declare -r NAMESPACE="${NAMESPACE:-}"
+
+# Environment file to store the ArgoCD crednetials.
+declare -r ARGOCD_ENV_FILE="${ARGOCD_ENV_FILE:-/rhtap/argocd/env}"
+
+fail() {
+    echo "# [ERROR] ${*}" >&2
+    exit 1
+}
+
+# Waits for the environment file to be available.
+wait_for_env_file() {
+    echo "# Waiting for '${ARGOCD_ENV_FILE}' to be available..."
+    for i in {1..30}; do
+        [[ -r "${ARGOCD_ENV_FILE}" ]] &&
+            return 0
+
+        wait=$((i * 15))
+        echo "### [${i}/30] Waiting for '${ARGOCD_ENV_FILE}' to be available..."
+        sleep ${wait}
+    done
+    return 1
+}
+
+# Stores the ArgoCD credentials in a Kubernetes secret.
+argocd_store_token() {
+    echo "# Asserting required environment variables..."
+    [[ -z "${SECRET_NAME}" ]] &&
+        fail "SECRET_NAME is not set!"
+    [[ -z "${NAMESPACE}" ]] &&
+        fail "NAMESPACE is not set!"
+
+    [[ -z "${ARGOCD_ENV_FILE}" ]] &&
+        fail "ARGOCD_ENV_FILE is not set!"
+
+    wait_for_env_file ||
+        fail "ARGOCD_ENV_FILE='${ARGOCD_ENV_FILE}' not found or not readable!"
+
+    echo "# Sourcing ArgoCD credentials from '${ARGOCD_ENV_FILE}'..."
+    # shellcheck source=/dev/null
+    . "${ARGOCD_ENV_FILE}"
+
+    [[ -z "${ARGOCD_HOSTNAME}" ]] &&
```

Comment:
Why not use `--from-file` and skip the validation?

Reply:
Interesting suggestion, I'll check that further.